src/main/java/ch/njol/skript/conditions/CondIsWearing.java

```diff
@@ -35,56 +11,68 @@
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.Kleenean;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.EntityEquipment;
+import org.bukkit.inventory.EquipmentSlot;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Is Wearing")
-@Description("Checks whether a player is wearing some armour.")
-@Examples({"player is wearing an iron chestplate and iron leggings",
-		"player is wearing all diamond armour"})
+@Description("Checks whether an entity is wearing some items (usually armor).")
+@Examples({
+	"player is wearing an iron chestplate and iron leggings",
+	"player is wearing all diamond armour",
+	"target is wearing wolf armor"
+})
 @Since("1.0")
 public class CondIsWearing extends Condition {
 	
 	static {
 		PropertyCondition.register(CondIsWearing.class, "wearing %itemtypes%", "livingentities");
 	}
 	
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Expression<LivingEntity> entities;
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Expression<ItemType> types;
 	
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] vars, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+	public boolean init(Expression<?>[] vars, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		entities = (Expression<LivingEntity>) vars[0];
 		types = (Expression<ItemType>) vars[1];
 		setNegated(matchedPattern == 1);
 		return true;
 	}
 	
 	@Override
-	public boolean check(final Event e) {
-		return entities.check(e,
-				en -> types.check(e,
-						t -> {
-							EntityEquipment equip = en.getEquipment();
-							if (equip == null)
-								return false; // No equipment -> not wearing anything
-							for (final ItemStack is : equip.getArmorContents()) {
-								if (t.isOfType(is) ^ t.isAll())
-									return !t.isAll();
-							}
-							return t.isAll();
-						}),
-				isNegated());
+	public boolean check(Event event) {
+		return entities.check(event, entity -> {
+			EntityEquipment equipment = entity.getEquipment();
+			if (equipment == null)
+				return false; // spigot nullability, no identifier as to why this occurs
+
+			ItemStack[] contents = Arrays.stream(EquipmentSlot.values())
+				.map(equipment::getItem)
+				.toArray(ItemStack[]::new);
+
+			return types.check(event, type -> {
```

Comment:
You should cache the values of `types` and use the static `check` method in `SimpleExpression` instead of evaluating the expression multiple times which can lead to unexpected results